app/controllers/course/video/submission/sessions_controller.rb

```diff
@@ -15,6 +15,8 @@ def update
                                   last_video_time: update_params[:last_video_time])
     end
     @session.merge_in_events!(update_params[:events])
+    # Update video duration using data from frontend VideoPlayer
+    @video.update_duration!(video_params[:video_duration]) if @video.duration < video_params[:video_duration]
```

Comment:
Wouldn't it be more straightforward to just do

```
@video.duration = video_params[:video_duration].round
```

Then the extra function in the video model won't be necessary also.

Reply:
ok can